perfkitbenchmarker/background_workload.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing classes for background workloads."""
+
+from perfkitbenchmarker import os_types
+from perfkitbenchmarker import vm_util
+
+BACKGROUND_WORKLOADS = []
+
+BACKGROUND_IPERF_PORT = 20001
+BACKGROUND_IPERF_SECONDS = 2147483647
+
+
+class AutoRegisterBackgroundWorkloadMeta(type):
+  """Metaclass which allows BackgroundWorkloads to be auto-registered."""
+
+  def __init__(cls, name, bases, dct):
+    super(AutoRegisterBackgroundWorkloadMeta, cls).__init__(name, bases, dct)
+    BACKGROUND_WORKLOADS.append(cls)
+
+
+class BaseBackgroundWorkload(object):
+  """Baseclass for background workloads."""
+  __metaclass__ = AutoRegisterBackgroundWorkloadMeta
+
+  EXCLUDED_OS_TYPES = []
+
+  @staticmethod
+  def IsEnabled(vm):
+    """Returns true if this background workload is enabled on this VM."""
+    return False
+
+  @staticmethod
+  def Prepare(vm):
+    """Prepares the background workload on this VM."""
+    pass
+
+  @staticmethod
+  def Start(vm):
+    """Starts the background workload on this VM."""
+    pass
+
+  @staticmethod
+  def Stop(vm):
+    """Stops the background workload on this VM."""
+    pass
+
+
+class CpuWorkload(BaseBackgroundWorkload):
+  """Workload that runs sysbench in the background."""
+
+  @staticmethod
+  def IsEnabled(vm):
+    """Returns true if this background workload is enabled on this VM."""
+    return bool(vm.OS_TYPE != os_types.WINDOWS and
```

Comment:
If someone does ask for background workload on a windows VM, it is silently ignored?